### PR TITLE
Add new column to Procedures table

### PR DESCRIPTION
**Pull request by kollil**
#### Rationale
Added new column to the ehr_lookups.procedures table to record the USDA pain categories. This field is not a required field.

#### Changes
* Update scripts
* Schema and metadata updates

### CODE REVIEW

**Review by Ohsudev — APPROVED** [review]

**Comment by kollil** [comment]
@labkey-jeckels , 
I am making changes to the core module. Added a new column (nullable) called, "PainCategories" to ehr_lookups.procedures table to record the USDA pain levels requested by the Vets. This new data will assist them when they have to produce an annual report of animal usage to the USDA. Please advise me if this addition is okay into the core. Otherwise, I will back off and create an ONPRC version of it.

Thanks
Lakshmi

**Comment by labkey-adam** [comment]
@labkey-jeckels and @labkey-martyp will review shortly. If we move forward with this approach then it looks like there needs to be a PostgreSQL version of the scripts and corrections to ehr_lookups.xml.

**Comment by labkey-keith** [comment]
@kollil As Adam mentioned, please add Josh and Marty as reviewers when you need to make a change to the EHR repository.

**Comment by kollil** [comment]
Sure @keith-labkey ,  I will not proceed until I hear from Josh or Marty. From now on, any core related changes, I will add both of them as reviewers.

**Review by labkey-martyp — CHANGES_REQUESTED** [review]
Hi Lakshmi, adding pain categories to procedures looks like a good update. We don't need to add the actual hard table ehr_lookups.Procedure_PainCategories. You can add new ehr_lookups sets directly to ehr_lookups.lookup_sets and ehr_lookups.lookups through the UI. I can help you with that. I'll remove that startup script from this PR and add the PG part for the procedures upgrade.

**Comment by kollil** [comment]
Thank you Marty, I agree with you. I don't need to add hard table for the pain levels. I will add the pain levels via UI.

**Comment by kollil** [comment]
@labkey-martyp , 
Please check the changes I made when you get a chance. The hard table has been deleted and the pain levels are created in the ehr_lookups.lookups and ehr_lookups.lookupsets. The changes are working fine and is installed on test server for user testing.

Thanks
Lakshmi

**Comment by labkey-martyp** [comment]
> @labkey-martyp , Please check the changes I made when you get a chance. The hard table has been deleted and the pain levels are created in the ehr_lookups.lookups and ehr_lookups.lookupsets. The changes are working fine and is installed on test server for user testing.
> 
> Thanks Lakshmi

@kollil Thanks for updating that. Since this is just in a feature branch we can just delete the upgrade script instead of dropping the table in another upgrade script. I just pushed a change that deletes the upgrade script and changes the names to match the convention of other ehr_lookups tables and columns. I'll let it run on TeamCity. Can you please verify if this will work for you. Thank you!

**Review by labkey-jeckels — COMMENTED** [review]
Looks good to me too.

**Comment by kollil** [comment]
Thanks @labkey-martyp, I got the update into my FB and installed the changes on our test server. Everything looks good.

**Comment by kollil** [comment]
@labkey-martyp , Please review the code and let me know if there are any more issues.

Thanks
Lakshmi

File: ehr/resources/queries/ehr_lookups/procedures.query.xml [hunk 1]
```diff
@@ -24,6 +24,16 @@
                             <fkColumnName>value</fkColumnName>
                         </fk>
                     </column>
+                    <!-- Kollil, 12/08/2022: Added a new column (drop down list) to display the USDA pain categories. -->
+                    <column columnName="painCategories">
+                        <columnTitle>Pain Categories</columnTitle>
+                        <fk>
+                            <fkDbSchema>ehr_lookups</fkDbSchema>
+                            <fkTable>Procedure_Pain_Categories</fkTable>
+                            <fkColumnName>rowid</fkColumnName>
```
Comment (labkey-martyp) [inline ehr/resources/queries/ehr_lookups/procedures.query.xml]: @kollil  We generally try not to use the rowId for lookup columns. Is there a reason you can't use the value?

Comment (kollil) [reply ehr/resources/queries/ehr_lookups/procedures.query.xml]: Sure. I have no particular reason to do that way. I saw similar code for a different field and followed the same thing. I can change it to "value".

Comment (labkey-martyp) [reply ehr/resources/queries/ehr_lookups/procedures.query.xml]: Would this work for you? Then the value doesn't have to be a long name. That can be the Title on the lookup.
`<column columnName="painCategories">
    <columnTitle>Pain Categories</columnTitle>
    <fk>
        <fkDbSchema>ehr_lookups</fkDbSchema>
        <fkTable>Procedure_Pain_Categories</fkTable>
        <fkColumnName>value</fkColumnName>
        <fkDisplayColumnName>title</fkDisplayColumnName>
    </fk>
</column>`

Comment (labkey-martyp) [reply ehr/resources/queries/ehr_lookups/procedures.query.xml]: Sorry formatting got kind of messed up on that

**Review by labkey-martyp — APPROVED** [review]
This looks good. I'll go ahead and merge this in. @kollil Please add titles to your values in ehr_lookups.Procedure_Pain_Categories.

**Pull request merged**